packages/search/src/elasticsearch/utils.ts

```diff
@@ -362,3 +373,54 @@ export async function getUser(practitionerId: string, authHeader: any) {
   })
   return await res.json()
 }
+
+function getPreviousStatus(body: IBirthCompositionBody) {
+  if (body.operationHistories && body.operationHistories.length > 0) {
+    return body.operationHistories[body.operationHistories.length - 1]
+      .operationType
+  }
+
+  return null
+}
+
+export function isValidOperationHistory(body: IBirthCompositionBody) {
+  const validStatusMapping = {
+    [IN_PROGRESS_STATUS]: [null],
```

Comment:
We observed that the time of the duplicate history is different than the original history in spite of being the same date. It can be reproduced by calling the same mutation twice or more as there had been no blocking previously. After a discussion with @mushrafulhoque-dsi , we have come to a solution that there can be a mapping of the statues and a set of valid statuses they can come from. (i.e. an application can only be `REGISTERED` from the beginning, `IN_PROGRESS`, `DECLARED`, `REJECTED`, `VALIDATED` or `WAITING_VALIDATION`, but not `REGISTERED`, or `CERTIFIED`) . If the history is not valid according to this mapping the operation history is not added

Reply:
Great spot @tahmidrahman-dsi .